source/MaterialXCore/Look.cpp

```diff
@@ -29,7 +29,7 @@ vector<MaterialAssignPtr> getGeometryBindings(ConstNodePtr materialNode, const s
         {
             if (matAssign->getReferencedMaterial() == materialNode)
             {
-                if (geomStringsMatch(geom, matAssign->getActiveGeom()))
+                if (geomStringsMatch(matAssign->getActiveGeom(), geom, true))
```

Comment:
The previous test a geometry path against the geometry string for an assignment so
a parent could pick up a child's assignment (e.g. geom: "/A" and could be assigned the material in assigned to: "/A/B").

This has now be reversed to test that the assignment is completely contained within the test geometry so an assignment to "/A/B" will not be assigned to a geometry "/A".

Reply:
Good catch!